elfo-network/src/socket/raw/tcp/metrics.rs

```diff
@@ -0,0 +1,314 @@
+use std::os::unix::io::AsRawFd;
+
+use metrics::gauge;
+
+#[derive(Debug, Default)]
+pub(crate) struct TcpMetrics {
+    /// Smoothed round trip time (microseconds)
+    pub rtt: u32,
+    /// RTT variance (microseconds)
+    pub rtt_var: u32,
+    /// Retransmission timeout (microseconds)
+    pub rto: u32,
+
+    /// Congestion window size
+    pub snd_cwnd: u32,
+    /// Slow start threshold
+    pub snd_ssthresh: u32,
+
+    /// Total retransmitted packets
+    pub retrans: u32,
+    /// Lost packets
+    pub lost: u32,
+    /// Unacknowledged packets
+    pub unacked: u32,
+    /// Selectively acknowledged packets
+    pub sacked: u32,
+
+    /// Path MTU
+    pub pmtu: u32,
+    /// Receive buffer space
+    pub rcv_space: u32,
+}
+
+pub(crate) fn collect_tcp_metrics<T: AsRawFd>(socket: &T) -> Result<TcpMetrics, std::io::Error> {
+    #[cfg(target_os = "linux")]
```

Comment:
The whole module is linux-only, why do we need extra check here?

Reply:
del